src/test/ui/rfc-2091-track-caller/tracked-trait-impls.rs

```diff
@@ -0,0 +1,40 @@
+// run-pass
+
+#![feature(track_caller)]
+
+/// Checks that the function this was called by is *also* caller-tracked.
+#[track_caller]
+fn check_caller_is_tracked(line_called_at: u32) {
+    let location = std::panic::Location::caller();
+    assert_eq!(location.file(), file!());
+    assert_ne!(location.line(), line_called_at, "line should be outside this fn");
+}
+
+trait Tracked {
+    fn local_tracked(&self);
+
+    #[track_caller]
+    fn blanket_tracked(&self);
+
+    #[track_caller]
+    fn default_tracked(&self) {
+        check_caller_is_tracked(line!());
+    }
```

Comment:
Can you add a second `impl` where you override this default method?

Reply:
There are two possible cases when overriding the default: with the attribute and without. Should both have the codegen flag?

Reply:
I'm going with "yes" since that's what happens by default with the codegen flag inheritance.